opennms-alarms/daemon/src/main/java/org/opennms/netmgt/alarmd/AlarmPersisterImpl.java

```diff
@@ -263,11 +269,31 @@ private void reduceEvent(OnmsEvent e, OnmsAlarm alarm, Event event) {
             }
         }
 
+        Set<OnmsAlarm> containedAlarms = getAlarms(event.getParmCollection(), m_alarmDao);
+        if (containedAlarms != null && !containedAlarms.isEmpty()) {
+            if (alarm instanceof Situation) {
+                for(OnmsAlarm related : containedAlarms) {
+                    // Related Alarms are additive for Situations reduced from Events.
+                    ((Situation)alarm).addAlarm(related);
+                }
+            } else {
+                LOG.warn("reduceEvent: Event {} attempts to add alarms to alarm that is not a Situation: {}.", event.getUei(), alarm);
```

Comment:
Would it be possible to change the type to a situation in this case? I could foresee cases where we wanted to associate alarms with an existing alarm rather than group these under a new situation.

Reply:
Can alarms be related to more than one situation?

Reply:
@pbrane Currently, yes. We do not explicitly prevent or constrain that behavior.

Reply:
@j-white Spent some time looking into conversion. It is tricky to convert on the fly to the sub-type. (We probably need to use a native query in a transaction).
If adding Alarms to existing alarms is the behavior we want, I think we would be best to remove the Alarm-Situation object hierarchy and just have relatedAlarms as an optional attribute of OnmsAlarm.

Reply:
Make sense.  Given that this is not easy to change, let's just keep the current behavior. If the use case is strong enough, we'll come back and revisit this.